src/js/main-section.js

```diff
@@ -0,0 +1,73 @@
+import Notiflix from 'notiflix';
+import { Loading } from 'notiflix/build/notiflix-loading-aio';
+
+import { fetchBooks } from './fetchBooks.js';
+import Notiflix from 'notiflix';
+
+allCategoryBtn.addEventListener('click', onAllCategoriesClick);
+
+function onAllCategoriesClick() {
+  startPage();
+}
+
+startPage();
+
+function markupBooks(array) {
+  let bestBooks;
+  const categories = array
+    .map(arr => {
+      bestBooks = arr.books
+        .map(({ _id, book_image, title, author }) => {
+          let placeholder;
+          const imgSrc = true ? `${book_image}` : placeholder;
+          if (arr.length === 0) {
```

Comment:
You check if an "arr" is empty after some of the code was executed. If "arr" is empty bestBooks will be still undefined after line 19. Its recommended to add "arr" verification at the beginning of the map method.